### ISSUE TITLE
Getting a NoMethodError when using MashapeAnalytics::Frameworks::Rack

### ISSUE THREAD

**Issue by mosic** [Mashape/analytics-agent-ruby#1]
Hi, I just tried using this library in our Rack application (using [Grape](https://github.com/intridea/grape)) and I can't get it to work. The app is [evercam-api](https://github.com/evercam/evercam-api), I made a branch for adding this: [mashape_analytics](https://github.com/evercam/evercam-api/compare/mashape_analytics).

Here is what I'm getting: 

```
NoMethodError: undefined method `size' for #<Rack::Lint::InputWrapper:0x000000073f0de8>
    /home/jack/.rvm/gems/ruby-2.2.0/gems/mashape-analytics-1.0.1/lib/mashape-analytics/frameworks/rack.rb:105:in `request_content_size'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/mashape-analytics-1.0.1/lib/mashape-analytics/frameworks/rack.rb:137:in `record_alf'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/mashape-analytics-1.0.1/lib/mashape-analytics/frameworks/rack.rb:41:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/newrelic_rpm-3.12.0.288/lib/new_relic/agent/instrumentation/middleware_tracing.rb:67:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/airbrake-4.2.1/lib/airbrake/rack.rb:46:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/newrelic_rpm-3.12.0.288/lib/new_relic/agent/instrumentation/middleware_tracing.rb:67:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/session/abstract/id.rb:225:in `context'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/session/abstract/id.rb:220:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/newrelic_rpm-3.12.0.288/lib/new_relic/agent/instrumentation/middleware_tracing.rb:67:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-cors-0.4.0/lib/rack/cors.rb:80:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/newrelic_rpm-3.12.0.288/lib/new_relic/agent/instrumentation/middleware_tracing.rb:67:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/bundler/gems/rack-ssl-enforcer-4e5bdac7af35/lib/rack/ssl-enforcer.rb:57:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/newrelic_rpm-3.12.0.288/lib/new_relic/agent/instrumentation/middleware_tracing.rb:67:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/urlmap.rb:66:in `block in call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/urlmap.rb:50:in `each'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/urlmap.rb:50:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/tempfile_reaper.rb:15:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/lint.rb:49:in `_call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/lint.rb:37:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/showexceptions.rb:24:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/commonlogger.rb:33:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:218:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/chunked.rb:54:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/content_length.rb:15:in `call'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/thin-1.6.3/lib/thin/connection.rb:86:in `block in pre_process'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/thin-1.6.3/lib/thin/connection.rb:84:in `catch'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/thin-1.6.3/lib/thin/connection.rb:84:in `pre_process'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/thin-1.6.3/lib/thin/connection.rb:53:in `process'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/thin-1.6.3/lib/thin/connection.rb:39:in `receive_data'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/eventmachine-1.0.7/lib/eventmachine.rb:187:in `run_machine'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/eventmachine-1.0.7/lib/eventmachine.rb:187:in `run'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/thin-1.6.3/lib/thin/backends/base.rb:73:in `start'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/thin-1.6.3/lib/thin/server.rb:162:in `start'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/handler/thin.rb:19:in `run'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/server.rb:286:in `start'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/lib/rack/server.rb:147:in `start'
    /home/jack/.rvm/gems/ruby-2.2.0/gems/rack-1.6.4/bin/rackup:4:in `<top (required)>'
    /home/jack/.rvm/gems/ruby-2.2.0/bin/rackup:23:in `load'
    /home/jack/.rvm/gems/ruby-2.2.0/bin/rackup:23:in `<main>'
    /home/jack/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `eval'
    /home/jack/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `<main>'
```

I also tried adding it to our Rails app [evercam-dashboard](https://github.com/evercam/evercam-dashboard) as a sanity check but I'm also getting errors there:

```
undefined method `size' for #<Puma::NullIO:0x00000002e85c18>

Extracted source (around line #105):

        request['HTTP_CONTENT_LENGTH'].to_i
      else
        request['rack.input'].size
      end
    end

mashape-analytics (1.0.1) lib/mashape-analytics/frameworks/rack.rb:105:in `request_content_size'
mashape-analytics (1.0.1) lib/mashape-analytics/frameworks/rack.rb:137:in `record_alf'
mashape-analytics (1.0.1) lib/mashape-analytics/frameworks/rack.rb:41:in `call'
warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
warden (1.2.3) lib/warden/manager.rb:34:in `catch'
warden (1.2.3) lib/warden/manager.rb:34:in `call'
rack (1.6.4) lib/rack/etag.rb:24:in `call'
rack (1.6.4) lib/rack/conditionalget.rb:25:in `call'
rack (1.6.4) lib/rack/head.rb:13:in `call'
actionpack (4.2.2) lib/action_dispatch/middleware/params_parser.rb:27:in `call'
actionpack (4.2.2) lib/action_dispatch/middleware/flash.rb:260:in `call'
rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
actionpack (4.2.2) lib/action_dispatch/middleware/cookies.rb:560:in `call'
activerecord (4.2.2) lib/active_record/query_cache.rb:36:in `call'
activerecord (4.2.2) lib/active_record/connection_adapters/abstract/connection_pool.rb:649:in `call'
actionpack (4.2.2) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
activesupport (4.2.2) lib/active_support/callbacks.rb:88:in `call'
activesupport (4.2.2) lib/active_support/callbacks.rb:88:in `_run_callbacks'
activesupport (4.2.2) lib/active_support/callbacks.rb:776:in `_run_call_callbacks'
activesupport (4.2.2) lib/active_support/callbacks.rb:81:in `run_callbacks'
actionpack (4.2.2) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
actionpack (4.2.2) lib/action_dispatch/middleware/reloader.rb:73:in `call'
actionpack (4.2.2) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
airbrake (4.2.1) lib/airbrake/rails/middleware.rb:13:in `call'
actionpack (4.2.2) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
actionpack (4.2.2) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
railties (4.2.2) lib/rails/rack/logger.rb:38:in `call_app'
railties (4.2.2) lib/rails/rack/logger.rb:20:in `block in call'
activesupport (4.2.2) lib/active_support/tagged_logging.rb:68:in `block in tagged'
activesupport (4.2.2) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (4.2.2) lib/active_support/tagged_logging.rb:68:in `tagged'
railties (4.2.2) lib/rails/rack/logger.rb:20:in `call'
quiet_assets (1.1.0) lib/quiet_assets.rb:27:in `call_with_quiet_assets'
actionpack (4.2.2) lib/action_dispatch/middleware/request_id.rb:21:in `call'
rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
rack (1.6.4) lib/rack/runtime.rb:18:in `call'
activesupport (4.2.2) lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
rack (1.6.4) lib/rack/lock.rb:17:in `call'
actionpack (4.2.2) lib/action_dispatch/middleware/static.rb:113:in `call'
heroku_rails_deflate (1.0.3) lib/heroku_rails_deflate/serve_zipped_assets.rb:58:in `call'
rack (1.6.4) lib/rack/deflater.rb:35:in `call'
rack (1.6.4) lib/rack/sendfile.rb:113:in `call'
airbrake (4.2.1) lib/airbrake/user_informer.rb:16:in `_call'
airbrake (4.2.1) lib/airbrake/user_informer.rb:12:in `call'
railties (4.2.2) lib/rails/engine.rb:518:in `call'
railties (4.2.2) lib/rails/application.rb:164:in `call'
rack (1.6.4) lib/rack/content_length.rb:15:in `call'
puma (2.11.3) lib/puma/server.rb:507:in `handle_request'
puma (2.11.3) lib/puma/server.rb:375:in `process_client'
puma (2.11.3) lib/puma/server.rb:262:in `block in run'
puma (2.11.3) lib/puma/thread_pool.rb:104:in `call'
puma (2.11.3) lib/puma/thread_pool.rb:104:in `block in spawn_thread'
```

Any ideas?

**Comment by SGrondin** [comment]
Pinging @kennethklee

**Comment by ahmadnassri** [comment]
@mosic can you try manually setting `size` to `0` temporarily while @kennethklee investigates.

**Comment by kennethklee** [comment]
Hmm looks like two issues. One with Rack::Lint::InputWrapper and one with Puma::NullIO. Neither have a size field. I'll have to figure out how to get the size from them another way.

**Comment by kennethklee** [comment]
@mosic Just released 1.0.2 with the fix. Please let me know if it still happens.

**Comment by mosic** [comment]
@kennethklee Thanks, updated and now getting a different error: https://github.com/Mashape/analytics-agent-ruby/issues/2
